### PR TITLE
Implement Ros2PublishRadarScanNode with tests

### PR DESCRIPTION
**Pull request by prybicki**

### CODE REVIEW

File: test/src/graph/nodes/Ros2PublishRadarScanNodeTest.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#include <helpers/sceneHelpers.hpp>
+#include <helpers/graphHelpers.hpp>
+#include <helpers/testPointCloud.hpp>
+
+#include <graph/NodesRos2.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+
+// TODO: replace with API calls
+#include <api/apiCommon.hpp>
```
Comment (msz-rai) [inline test/src/graph/nodes/Ros2PublishRadarScanNodeTest.cpp:11]:
```suggestion
#include <rgl/api/extensions/ros2.h>

#include <rclcpp/rclcpp.hpp>
#include <radar_msgs/msg/radar_scan.hpp>
```
Seems to be already resolved.

File: test/src/graph/nodes/Ros2PublishRadarScanNodeTest.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+TEST(Ros2PublishRadarScanNodeTest, should_receive_sent_data)
```
Comment (msz-rai) [inline test/src/graph/nodes/Ros2PublishRadarScanNodeTest.cpp:13]:
```suggestion
class Ros2PublishRadarScanNodeTest : public RGLTest {};

TEST_F(Ros2PublishRadarScanNodeTest, should_receive_sent_data)
```
You should define `Ros2PublishRadarScanNodeTest` class and derive it from `RGLTest` to set the logging configuration at the test startup and call `rgl_cleanup` at the test exit.
Also, it should be a Test Fixture.

File: test/src/graph/nodes/Ros2PublishRadarScanNodeTest.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		auto start = std::chrono::steady_clock::now();
+		do {
+			rclcpp::spin_some(node);
+			std::this_thread::sleep_for(std::chrono::milliseconds(10));
+		} while (messageCount != MESSAGE_REPEATS &&
+		         std::chrono::steady_clock::now() - start < std::chrono::seconds(WAIT_TIME_SECS));
+		ASSERT_EQ(messageCount, MESSAGE_REPEATS);
+	}
+
+	rclcpp::shutdown();
```
Comment (msz-rai) [inline test/src/graph/nodes/Ros2PublishRadarScanNodeTest.cpp:72]: No needed if RGL initialized `rclcpp`. `Ros2InitGuard` did it when creating the ROS2 publishing node. `rclcpp` will be shut down automatically when cleaning up.

**Review by msz-rai — CHANGES_REQUESTED** [review]
Good job :+1:
I have left some comments to address.

File: extensions/ros2/src/api/apiRos2.cpp [hunk 4]
```diff
@@ -83,12 +83,38 @@ RGL_API rgl_status_t rgl_node_points_ros2_publish_with_qos(rgl_node_t* node, con
...
+                                                     rgl_qos_policy_history_t qos_history, int32_t qos_history_depth)
+{
+	auto status = rglSafeCall([&]() {
+		RGL_DEBUG("rgl_node_publish_ros2_radarscan(node={}, topic_name={}, frame_id={})", repr(node), topic_name, frame_id);
+		CHECK_ARG(topic_name != nullptr);
+		CHECK_ARG(topic_name[0] != '\0');
+		CHECK_ARG(frame_id != nullptr);
+		CHECK_ARG(frame_id[0] != '\0');
+
+		createOrUpdateNode<Ros2PublishRadarScanNode>(node, topic_name, frame_id);
```
[the remark this answers is not in the document]
Comment (prybicki) [reply extensions/ros2/src/api/apiRos2.cpp:107]: Fixed + removed default params.

File: extensions/ros2/src/api/apiRos2.cpp [hunk 5]
```diff
@@ -83,12 +83,38 @@ RGL_API rgl_status_t rgl_node_points_ros2_publish_with_qos(rgl_node_t* node, con
...
+	});
+	TAPE_HOOK(node, topic_name, frame_id);
+	return status;
+}
+
+void TapeRos2::rgl_node_publish_ros2_radarscan(const YAML::Node& yamlNode, PlaybackState& state)
+{
+	auto nodeId = yamlNode[0].as<TapeAPIObjectID>();
+	rgl_node_t node = state.nodes.contains(nodeId) ? state.nodes[nodeId] : nullptr;
+	rgl_node_points_ros2_publish(&node, yamlNode[1].as<std::string>().c_str(), yamlNode[2].as<std::string>().c_str());
```
[the remark this answers is not in the document]
Comment (prybicki) [reply extensions/ros2/src/api/apiRos2.cpp:117]: And the function being called is wrong ;)

**Review by msz-rai — APPROVED** [review]
LGTM

**Pull request merged**